src/app/cli/src/commands/system_diagnose_command.rs

```diff
@@ -281,11 +281,11 @@ impl DiagnosticCheck for CheckWorkspaceConsistent {
         });
         let datasets: Vec<_> = self.dataset_repo.get_all_datasets().try_collect().await?;
 
-        let verify_options = VerificationOptions {
+        let verify_options = Arc::new(VerificationOptions {
```

Comment:
This insertion of `Arc` all over the interface is a wrong idea.
The problem you've been trying to solve with the task spawning can be expressed much easier - clone options locally before spawn callback:

```
 let handles: Vec<_> = requests
            .into_iter()
            .map(|r| {
                ...
                let clone_options = options.clone();
```